### ISSUE TITLE
Symfony 2.7 with 3.0 directory structure

### ISSUE THREAD

**Issue by zulus** [symfony/symfony-installer#174]
In deprecated install method I'm able to use `SENSIOLABS_ENABLE_NEW_DIRECTORY_STRUCTURE=true`.

Is there any way to do the same with symfony installer?

**Comment by Pierstoval** [comment]
Actually no, you must do it manually, or with the composer installation.
 
I think it's not entirely right to say "deprecated" install, it's just "less recommended" than the official installer :wink:

**Comment by zulus** [comment]
This article say that composer it's deprecated: http://symfony.com/blog/introducing-the-new-symfony-installer

Anyway thanks for fast response. Would be nice to have option --sf3structure in installer.

**Comment by Pierstoval** [comment]
An `--sf3structure` option does not seem to be possible (or implies maany actions from the installer). 
Actually, the Symfony standard edition is downloaded from `get.symfony.com` as a zipball, generated right on the server based on the Standard Edition repo (and many other things that depends on SensioLabs). The installer only downloads this archive, unzip it, clean it (remove "UPGRADE" files, replace the README, etc.), change the composer.json file to have your username and app name replacing Symfony's one, etc.
Changing the directory structure, as it is not standard yet, might lead to issues.
 
The SF3 structure will obviously be available once the SF3 branch will reach an "installable" state, so we'll need to wait a little :wink:

**Comment by javiereguiluz** [comment]
@zulus although we might change our mind in the future, right now we think that Symfony 2 apps should use the traditional structure and Symfony 3 apps should use the new structure. That's why there is no option to change this behavior. See https://github.com/symfony/symfony-installer/issues/93 for a related discussion.

So for now I'm closing this issue as "won't fix", but we'll revisit it again in the future. Thanks!

**Comment by c33s** [comment]
please reopen, now that symfony3 is live, it sould be possible to use the new directory structure with symfony 2.8 (also having an option for 2.7 would be nice)

**Comment by stof** [comment]
@c33s I don't see how the fact that Symfony 3 is released should change the policy for Symfony 2

**Comment by c33s** [comment]
@stof because if i have to stick with sf2 to wait until all bundles are compatible with sf3, i would currently start a project with sf2.8. but to quickly be able to upgrade to 3.0 i would like to use the new directory structure.

**Comment by Pierstoval** [comment]
Why wouldn't you upgrade the structure when you migrate to Sf3 ?

**Comment by stof** [comment]
@c33s then just perform the changes manually, doing the structure upgrade just after creating the project. Changes are not hard: https://github.com/symfony/symfony-standard/compare/v2.8.0...v3.0.0 (note that some changes in this diff are not relevant, for instance moving from ``~`` to ``^`` in the composer.json is unrelated to the structure)

Generating 2 archives for each release to be able to switch between them would be a huge work, and would require newcomers to understand such choice.

**Comment by c33s** [comment]
why do you make it so difficult for developers to use the new directory stucture?

why i don't to upgrade when i migrate? because the new stucture is much better and i prefer to use it now. 

if you want to prevent newcommers from being confronted with the new structure, you can work with the good old environment variable, only if set give the option for the new structure.

also it helps newcommers to immediately use the new directory structure, so they get used to it.
i notice it by myself, switching from project to project having different structures is quite cumbersome.

after introducing the function with 2.5 (to choose the directory stucture) it is simply wrong to remove it with 2.7.5 and reintroduce it with 3.0. 

its only an option, let the user decide.

**Comment by Pierstoval** [comment]
@c33s Yeah, switching projects having different structures is cumbersome, and yes the new structure is better (this is why it's been implemented).
 
The problem is that the new structure is only for the new structure. I don't know what the team thinks about this, but I personally think that adopting a new structure for Symfony 2.x was wrong because it breaks BC in some bundles. As Symfony 3 is ought to break many BCs that were notified since Symfony 2.3, the structure is also a BC break so this *is* a problem to induce it in the 2.x branches

**Comment by stof** [comment]
@c33s introducing this option at the 2.5 time was a mistake: it caused a lot of confusion for newcomers, made it more difficult for the doc (as the doc cannot use both structures) and was buggy. This is why we removed the choice. Adding an option is precisely what confused newcomers, as they don't understand it.

And given that the installer downloads a pre-built archive, it is not *just an option*. It forces to generate 2 archives for each release

**Comment by c33s** [comment]
i really understand that it may be difficult for newcommers, its clear, that people get confused, that the docs don't match and that some bundles require the console residing in the `app`. 

also i know that the deployment of the new directory structure was a mistake but it happend. you can't put it back in the box. yes it is a BC break but it happend, taken it back in a patch version is feels like a double BC break. 
as the feature popped up, our company was happily switching to the new directory stucture with 2.5, if we start a new project, we use the new directory structure (and it is really cumbersome to create a new project with the new directory structure). see https://stackoverflow.com/a/34026087/590247
other developers and companies out there also use the new structure. they may not call out for help like the newcommers which where confused, they solve bundle breaks with symlinks of something like that, they make manual changes to the code, to force the new directory stucture. google a little bit, they are there and ask for the new directory stucture in sf2. 

a lot of bundles aren't compatible with symfony3 yet, so it is not possible to simply switch to sf3, a lot of people need to stick with sf2. 

the argument of having to maintain 2 builds in parallel doesn't really count for me, its like a 32bit and a 64bit version, like supporting different processortypes (imagine debian says ok no more 32bit builds and lets drop the support for all processors but amd64). 

its all integrateable in the build server, you only have to touch it once to create two different builds, otherwise really a lot of people out there have to manually change the directory structure. again and again. project for project.

don't only think about newcommers, there are also pro's out there who rely on functionality which was integrated with 2.5 (intentional  or unintentional)

to be consistent, the symfony team would have to create a new version of 2.5 and 2.6 and remove the question

the decision to remove the question was right but removing the option with the environment variable (guys who set this variable should know what they are doing), was wrong in my eyes.

so please, you really make the developers live easier.

**Comment by stof** [comment]
@c33s removing it was not a BC break, as this option was only available when creating new projects, and so it was not breaking any existing project updating their version.

New versions of 2.5 and 2.6 won't be done, as they are EOLed.
